react/features/base/label/components/CircularLabel.native.js

```diff
@@ -11,34 +11,106 @@ import styles from './styles';
 
 type Props = AbstractProps & {
 
+    /**
+     * Extra behaviour of the label. This prop adds some additional styles based
+     * on its value. E.g. if extraBehaviour = OFF, it will render the label
+     * symbolising that the thing it displays (e.g. recording) is off.
+     */
+    extraBehaviour: Symbol,
+
     /**
      * Style of the label.
      */
     style?: ?StyleType
 };
 
+type State = {
+
+    /**
+     * An animation object handling the opacity changes of the in progress
+     * label.
+     */
+    pulseAnimation: Object
+}
+
 /**
  * Renders a circular indicator to be used for status icons, such as recording
  * on, audio-only conference, video quality and similar.
  */
-export default class CircularLabel extends AbstractCircularLabel<Props> {
+export default class CircularLabel extends AbstractCircularLabel<Props, State> {
+    /**
+     * Extra behaviour that this label supports.
+     */
+    static behaviour = {
+        IN_PROGRESS: Symbol('CIRCULAR_LABEL_IN_PROGRESS'),
+        OFF: Symbol('CIRCULAR_LABEL_OFF')
+    };
+
+    /**
+     * Instantiates a new instance of {@code CircularLabel}.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            pulseAnimation: new Animated.Value(0)
+        };
+    }
+
+    /**
+     * Implements {@code React#componentDidMount}.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        const { pulseAnimation } = this.state;
+
+        Animated.loop(Animated.sequence([
+            Animated.timing(pulseAnimation, {
+                delay: 500,
+                toValue: 1,
+                useNativeDriver: true
+            }),
+            Animated.timing(pulseAnimation, {
+                toValue: 0.3,
+                useNativeDriver: true
+            })
+        ])).start();
+    }
+
     /**
      * Implements React {@link Component}'s render.
      *
      * @inheritdoc
      */
     render() {
-        const { label, style } = this.props;
+        const { extraBehaviour, label, style } = this.props;
+
+        let extraStyle = null;
+
+        switch (extraBehaviour) {
+        case CircularLabel.behaviour.IN_PROGRESS:
+            extraStyle = {
+                opacity: this.state.pulseAnimation
```

Comment:
shouldn't we skip initializing the animation if the behavior is not IN_PROGRESS, since it won't be used at all?

Reply:
It takes quite some extra steps to do that, but good point, thanks! I just refactored it, please review the new code